MCP server fails to start: Cannot find module '../../package.json'

## Bug Report

### Version
ai-os v0.22.0 (bundled MCP server)

### Environment
- OS: Windows 11
- Node.js installed at `C:\Program Files\nodejs\node.exe`
- Install path: `.github/ai-os/mcp-server/index.js`

### Description
The MCP server fails to start immediately after install/refresh with:

```
[ai-os:mcp] Fatal error: Cannot find module '../../package.json'
Require stack:
- C:\GitHub\UNIS-Frontend\.github\ai-os\mcp-server\index.js
```

### Root Cause
In the bundled `index.js`, the `createSdkServer()` function does:

```js
const req = createRequire(import.meta.url);
const pkgVersion = req("../../package.json").version;
```

The MCP server is installed at `.github/ai-os/mcp-server/index.js`.  
Resolving `../../package.json` from that path yields `.github/package.json` — which does **not** exist in a standard repo.

The project's actual `package.json` is at the repository root, three levels up (`../../../package.json`).

### Workaround
Manually create `.github/package.json` with a minimal version field:

```json
{
  "name": "ai-os",
  "version": "0.22.0"
}
```

This unblocks the server but the file is not created by the installer.

### Expected Behavior
Either:
- The installer creates `.github/package.json` automatically, **or**
- The bundle resolves the version from its own `mcp-server/package.json` (which already exists and has `version: "0.1.0"`), **or**
- The relative `require` path is corrected to `../../../package.json` (repo root)

### Steps to Reproduce
1. Fresh install or `--refresh-existing` on any repo where `.github/package.json` does not exist
2. Attempt to start the MCP server: `node .github/ai-os/mcp-server/index.js`
3. Observe fatal error: `Cannot find module '../../package.json'`

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "ai-os",
-  "version": "0.22.0",
+  "version": "0.22.1",
   "description": "Portable AI OS — scans any repo and generates optimized GitHub Copilot context, agents, skills, MCP tools, and slash-command prompts",
   "type": "module",
   "engines": {

--- a/src/mcp-server/sdk-server.ts
+++ b/src/mcp-server/sdk-server.ts
@@ -11,7 +11,6 @@ import { McpServer } from '@modelcontextprotocol/sdk/server/mcp.js';
 import { StdioServerTransport } from '@modelcontextprotocol/sdk/server/stdio.js';
 import { z } from 'zod';
 import path from 'node:path';
-import { createRequire } from 'node:module';
 import {
   getProjectRoot,
   readAiOsFile,
@@ -49,6 +48,7 @@ import {
   searchSymbols,
   getFilePurpose,
 } from './utils.js';
+import { resolveMcpServerVersion } from './shared.js';
 import { detectDrift, formatDriftReport } from '../detectors/drift.js';
 import { readFile, listDirectory, runTests, runLint, runBuild } from './filesystem.js';
 import { listWorkflows, loadWorkflow, validateWorkflow, buildWorkflowRunPlan, formatRunPlan } from '../workflow-runner.js';
@@ -72,8 +72,7 @@ function wrap(
 }
 
 export function createSdkServer(): McpServer {
-  const req = createRequire(import.meta.url);
-  const pkgVersion = (req('../../package.json') as { version: string }).version;
+  const pkgVersion = resolveMcpServerVersion(import.meta.url);
 
   const server = new McpServer({ name: 'ai-os', version: pkgVersion });
 

--- a/src/mcp-server/shared.ts
+++ b/src/mcp-server/shared.ts
@@ -5,6 +5,7 @@
  */
 import fs from 'node:fs';
 import path from 'node:path';
+import { fileURLToPath } from 'node:url';
 
 // ── Project root ───────────────────────────────────────────────────────────────
 
@@ -87,6 +88,37 @@ export function ensureSessionMemoryStore(): void {
   if (!fs.existsSync(failurePath)) {
     fs.writeFileSync(failurePath, '', 'utf-8');
   }
+}
+
+// ── Bundle version resolution ─────────────────────────────────────────────────
+
+/**
+ * Resolves the AI OS bundle version for use in the installed MCP server.
+ *
+ * When the bundle runs from `.github/ai-os/mcp-server/index.js`, relative paths
+ * like `../../package.json` incorrectly resolve to `.github/package.json`.
+ * Instead, we read `runtime-manifest.json` (co-located with the bundle) which
+ * the installer always writes with the correct `sourceVersion`. Falls back to
+ * the development-time `../../package.json` path and then to `'0.0.0'`.
+ */
+export function resolveMcpServerVersion(importMetaUrl: string): string {
+  const dir = path.dirname(fileURLToPath(importMetaUrl));
+
+  try {
+    const manifest = JSON.parse(
+      fs.readFileSync(path.join(dir, 'runtime-manifest.json'), 'utf-8'),
+    ) as { sourceVersion?: string };
+    if (manifest.sourceVersion) return manifest.sourceVersion;
+  } catch { /* not installed yet — fall through to dev path */ }
+
+  try {
+    const pkg = JSON.parse(
+      fs.readFileSync(path.join(dir, '..', '..', 'package.json'), 'utf-8'),
+    ) as { version?: string };
+    return pkg.version ?? '0.0.0';
+  } catch { /* dev package.json not found */ }
+
+  return '0.0.0';
 }
 
 // ── I/O utilities ──────────────────────────────────────────────────────────────

--- a/src/mcp-server/utils.ts
+++ b/src/mcp-server/utils.ts
@@ -9,7 +9,7 @@ import fs from 'node:fs';
 import path from 'node:path';
 import { fileURLToPath } from 'node:url';
 import { getLatestResolvableVersion } from '../updater.js';
-import { ROOT, readAiOsFile as _readAiOsFile } from './shared.js';
+import { ROOT, readAiOsFile as _readAiOsFile, resolveMcpServerVersion } from './shared.js';
 
 const __dirname = path.dirname(fileURLToPath(import.meta.url));
 
@@ -89,11 +89,8 @@ export function checkForUpdates(): string {
 
   let toolVersion = '0.0.0';
   try {
-    const toolPkg = JSON.parse(
-      fs.readFileSync(path.join(__dirname, '..', '..', 'package.json'), 'utf-8'),
-    ) as { version?: string };
-    toolVersion = toolPkg.version ?? '0.0.0';
-  } catch { /* tool package.json not found */ }
+    toolVersion = resolveMcpServerVersion(import.meta.url);
+  } catch { /* ignore */ }
 
   const latestVersion = getLatestResolvableVersion(toolVersion);
 

